internal/controllers/submariner/loadbalancer_resources.go

```diff
@@ -110,8 +111,14 @@ func newLoadBalancerService(instance *v1alpha1.Submariner, platformTypeOCP strin
 		if instance.Spec.HostedCluster {
 			externalTrafficPolicy = corev1.ServiceExternalTrafficPolicyTypeCluster
 		}
-	default:
-		svcAnnotations = map[string]string{}
+	}
+
+	if instance.Spec.LoadBalancerServiceAnnotations != nil {
+		if svcAnnotations == nil {
+			svcAnnotations = map[string]string{}
```

Comment:
I didn’t realise yesterday but we know the intended capacity for this:

```suggestion
			svcAnnotations = make(map[string]string, len(instance.Spec.LoadBalancerServiceAnnotations))
```